src/protocols/coda_pow.ml

```diff
@@ -745,6 +748,11 @@ module type Staged_ledger_base_intf = sig
 
   val create : ledger:ledger -> t
 
+  val unsafe_of_scan_state_and_ledger :
+    ledger:ledger -> scan_state:Scan_state.t -> t
+  (** Only call this when you're sure ledger and scan_state are correct
```

Comment:
We should continue using the unsafe variant trick, for instance, make it return the result in a `Unsafe` or something.